infobip_channels/sms/models/body/core.py

```diff
@@ -0,0 +1,130 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional, Union
+
+from pydantic import (
+    AnyHttpUrl,
+    Field,
+    StrictBool,
+    conint,
+    conlist,
+    constr,
+    root_validator,
+    validator,
+)
+
+from infobip_channels.core.models import CamelCaseModel, DateTimeValidator
+
+MINIMUM_DELIVERY_WINDOW_MINUTES = 60
+
+
+class ContentTypeEnum(str, Enum):
+    APPLICATION_JSON = "application/json"
+    APPLICATION_XML = "application/xml"
+
+
+class TrackEnum(str, Enum):
+    SMS = "SMS"
+    URL = "URL"
+
+
+class TypeEnum(str, Enum):
+    ONE_TIME_PIN = "ONE_TIME_PIN"
+    SOCIAL_INVITES = "SOCIAL_INVITES"
+
+
+class TimeUnitEnum(str, Enum):
+    MINUTE = "MINUTE"
+    HOUR = "HOUR"
+    DAY = "DAY"
+
+
+class DaysEnum(str, Enum):
+    MONDAY = "MONDAY"
+    TUESDAY = "TUESDAY"
+    WEDNESDAY = "WEDNESDAY"
+    THURSDAY = "THURSDAY"
+    FRIDAY = "FRIDAY"
+    SATURDAY = "SATURDAY"
+    SUNDAY = "SUNDAY"
+
+
+class IndiaDlt(CamelCaseModel):
+    content_template_id: Optional[str] = None
+    principal_entity_id: str
+
+
+class Regional(CamelCaseModel):
+    india_dlt: Optional[IndiaDlt] = None
+
+
+class Time(CamelCaseModel):
+    hour: conint(ge=0, le=23)
+    minute: conint(ge=0, le=59)
+
+
+class Destination(CamelCaseModel):
+    message_id: Optional[str] = None
+    to: constr(min_length=0, max_length=50)
+
+
+class DeliveryTimeWindow(CamelCaseModel):
+    days: conlist(DaysEnum, min_items=1)
+    from_time: Optional[Time] = Field(alias="from", default=None)
+    to: Optional[Time] = None
+
+    @root_validator
+    def validate_from_and_to(cls, values):
```

Comment:
I believe this is the same method as we have in the `mms` channel. We can extract it (along with `_validate_time_differences`) into a `FromAndToTimeValidator` and reuse it in both places.

Reply:
yea, good chunk of code is gone